include/crpropa/Grid.h

```diff
@@ -262,6 +271,16 @@ class Grid: public Referenced {
 	  By default this it the trilinear interpolation. The user can change the
 	  routine with the setInterpolationType function.*/
 	T interpolate(const Vector3d &position) {
+	// check for volume
+	if(clipVolume) {
+		Vector3d edge = origin + Vector3d(Nx, Ny, Nz) * spacing;
+		bool isInVolume = (position.x >= origin.x) && (position.x <= edge.x);
+		isInVolume &= (position.y >= origin.y) && (position.y <= edge.y);
+		isInVolume &= (position.z >= origin.z) && (position.z <= edge.z);
+		if(!isInVolume) 
+			return T(0.);
```

Comment:
Would a KISS message help the user to see that the current position is outside the grid?

Reply:
I would not do it. In a full simulation where it is possible that often the interpolation is called outside a grid that causes a lot of writing to the console which makes the simulation extremely slow and other important warnings can easily be overseen.